### PR TITLE
ENH: Adds Beta Rarefaction Visualizer

### PR DESCRIPTION
**Pull request by Kleptobismol**
Adds a visualizer to q2-diversity which includes two tabs:

Heatmap:
A heatmap representing the similarity of multiple rarefied distance matrices.
screen shot 2017-08-07 at 4 19 16 pm

Master Tree:
A download link for a newick tree file with confidence values for each node displayed. 
screen shot 2017-08-07 at 4 20 15 pm

These visualizations are computed by rarefying as many distance matrices as determined by the user, and comparing those matrices.

### CODE REVIEW

**Comment by thermokarst** [comment]
Keeping the 'DO-NOT-MERGE' label on this because I would like to see https://github.com/qiime2/q2-diversity/pull/128 make it in first, that way we can handle any merge conflicts in this PR, instead of putting that on @maxvonhippel to deal with in https://github.com/qiime2/q2-diversity/pull/128.

File: q2_diversity/plugin_setup.py [hunk 1]
```diff
@@ -351,3 +351,39 @@
...
+
+color_schemes = [
+    'BrBG', 'BrBG_r', 'PRGn', 'PRGn_r', 'PiYG', 'PiYG_r',
+    'PuOr', 'PuOr_r', 'RdBu', 'RdBu_r', 'RdGy', 'RdGy_r',
+    'RdYlBu', 'RdYlBu_r', 'RdYlGn', 'RdYlGn_r']
+
+plugin.visualizers.register_function(
+    function=q2_diversity._beta._visualizer.beta_rarefaction,
+    inputs={'table': FeatureTable[Frequency], 'phylogeny': Phylogeny[Rooted]},
+    parameters={'sampling_depth': Int,
```
Comment (ebolyen) [inline q2_diversity/plugin_setup.py]: `Int % Range(1, None)`

File: q2_diversity/plugin_setup.py [hunk 2]
```diff
@@ -351,3 +351,39 @@
...
+                'metric': Str % Choices(beta.all_metrics()),
+                'num_iterations': Int,
```
Comment (ebolyen) [inline q2_diversity/plugin_setup.py]: `Int % Range(1, None)`

**Comment by jairideout** [comment]
Travis failures are unrelated. This repo, along with several others, rely on upstream changes to `q2templates` that haven't made it into the staging environment files yet due to busywork's failing pipeline. Merge away!

**Pull request merged**